clang-tools-extra/clang-tidy/readability/QualifiedAutoCheck.cpp

```diff
@@ -174,6 +176,21 @@ void QualifiedAutoCheck::registerMatchers(MatchFinder *Finder) {
 
 void QualifiedAutoCheck::check(const MatchFinder::MatchResult &Result) {
   if (const auto *Var = Result.Nodes.getNodeAs<VarDecl>("auto")) {
+    if (RespectOpaqueTypes) {
+      auto DeducedType =
```

Comment:
Please do not use `auto` if type is not spelled explicitly in same statement or iterator.